src/realm.h

```diff
@@ -3088,6 +3102,43 @@ RLM_API void realm_sync_config_set_recovery_directory_path(realm_sync_config_t*,
 RLM_API void realm_sync_config_set_resync_mode(realm_sync_config_t*,
                                                realm_sync_session_resync_mode_e) RLM_API_NOEXCEPT;
 
+// flx sync c-api
+RLM_API realm_flx_sync_subscription_set_t* realm_sync_get_latest_subscription_set(const realm_t* realm);
+RLM_API realm_flx_sync_subscription_set_t* realm_sync_get_active_subscription_set(const realm_t* realm);
+RLM_API realm_flx_sync_mutable_subscription_set_t*
+realm_sync_get_mutable_subscription_set(const realm_flx_sync_subscription_set_t* subscription_set);
```

Comment:
I think we don't need to expose this in the C-API for now.

Reply:
OK got it, you are proposing to hide the conversation from subscription to mutable subscription inside the insertion methods, got it...

Reply:
At this point, we don't need to expose the mutable subscription class either…

Reply:
Oh, no, you do need the mutable subscription class. Sorry, I wasn't clear there. Users create mutable subscription sets when they call `make_mutable_copy()` off of a `SubscriptionSet`. Then you can call any of the mutation methods on the return value of that and eventually call `commit()` to commit it and have it sent to the server. Each time you call `make_mutable_copy()` you're creating a brand new subscription set in its own write transaction that must either be committed or aborted (by not calling commit before destroying the MutableSubscriptionSet). This specific function `get_mutable_subscription_set()` is used by the sync client to update Subscription Set states after processing download messages.

Reply:
OK, so I am confused on this. We need an interface for converting a subscription into a mutable one, then. On the same time, we do need to expose some interface for "doing a conversation".